Lat long alt plotter

AnikethL opened the pull request:
Plots a waypoint based on text input data

AnikethL committed:
Lat lon alt thing

AnikethL committed:
Lat lon alt thing

AnikethL committed:
Update FlightPlanToolbar

Lat lon alt plot point

AnikethL committed:
Update FlightPlanToolbar

Lat lon alt plot point

pizzalemon commented on client/src/pages/FlightData/tabs/FlightPlan/FlightPlanToolbar.js:
```diff
@@ -71,6 +91,111 @@ const FlightPlanToolbar = props => {
 
+	const addWaypoint = (lat, lon, alt) => {
+		// Create a new waypoint
+		console.log(parseInt(alt))
```
remove

pizzalemon commented on client/src/pages/FlightData/tabs/FlightPlan/FlightPlanToolbar.js:
```diff
@@ -71,6 +91,111 @@ const FlightPlanToolbar = props => {
+			return
+		}
+		else if (props.getters.placementType=="unlimLoiter"){
+			cmds=Commands.unlimLoiter
```
is there a point in defining this then setting cmd to this? if not, I don't really see a reason for it

pizzalemon commented on client/src/pages/FlightData/tabs/FlightPlan/FlightPlanToolbar.js:
```diff
@@ -71,6 +91,111 @@ const FlightPlanToolbar = props => {
+		}
+		
+		
+		console.log(props.getters.placementType)
```
should probably remove too

pizzalemon commented on client/src/pages/FlightData/tabs/FlightPlan/FlightPlanToolbar.js:
```diff
@@ -150,6 +275,91 @@ const FlightPlanToolbar = props => {
+						placeholder={greyOutPlacePoint() ? "Lat":"Disabled"}
+						content={greyOutPlacePoint() ? lat:""}
+						onChange={(e)=>setLat(e)}
+						style={{"backgroundColor":"#E2DBD5","borderColor":"transparent" }}
```
It's probably a good idea to write a separate component for these boxes, since they're the same thing copy pasted multiple times. I would recommend that. You can put the component in the same file, somewhere below the main component (FlightPlanToolbar).

pizzalemon reviewed (changes requested):
The waypoints created using the table need to be marked as unsaved (i.e. the waypoints being transparent when you place them using a mouse to signify that they're not saved to the flight file).

Also please follow the same style of coding as the rest of the Ground Station. There are a lot of random newlines, spaces, etc., as well as places where there should be a space which is missing. Please fix that.

AnikethL committed:
Update FlightPlanToolbar.js

AnikethL committed:
Update FlightPlanToolbar.js

AnikethL committed:
Lat_long_alt_plotter

AnikethL committed:
Lat_long_alt_plotter

AnikethL reviewed (commented):
Fixed the bugs

pizzalemon commented on client/src/pages/FlightData/tabs/FlightPlan/FlightPlanToolbar.js:
```diff
@@ -92,82 +92,27 @@ const FlightPlanToolbar = props => {
-				p4:0
-	
-			};
+			point = {alt:alt!="" ? parseFloat(alt):props.getters.defaultAlt, cmd:Commands.waypoint, lat:parseFloat(lat), lng:parseFloat(lon), num:props.getters.path.length + 1, p1:0, p2:0, p3:0, p4:0, opacity:0.5};
```
this should be expanded out

pizzalemon commented on client/src/pages/FlightData/tabs/FlightPlan/FlightPlanToolbar.js:
```diff
@@ -277,83 +222,69 @@ const FlightPlanToolbar = props => {
+						onChange={(e) => setLat(e)} 
+						style={{"backgroundColor":"#E2DBD5","borderColor":"transparent" }} 
+						onKeyDown={(e) => enterPlacePoint(e)}
 					/>
```
you're missing the point of wrapping this within its own component. you don't have to pass all those props down. they're redundant, and can just be put inside the component itself.

AnikethL committed:
Insert Waypoints + Table

Cleaned up the code and also added a table of the waypoints

AnikethL committed:
Insert Waypoints + Table

Cleaned up the code and also added a table of the waypoints

pizzalemon committed:
Fix formatting

pizzalemon reviewed (dismissed):
It could look nicer, but it's functionally complete.

pizzalemon commented:
pushed it to dev

AnikethL committed:
Merge branch 'main' of github.com:AnikethL/GroundStation

AnikethL committed:
Added Edit functionality in table

- Added edit+delete functionality from table
- added settings gear icon for connection button
- added waypoint highlighting on map when point number clicked
